pyaptamer/experiments/_aptamer_eval.py

```diff
@@ -8,8 +10,92 @@
 from pyaptamer.utils import encode_rna, rna2vec
 
 
-class Aptamer(BaseObject):
-    """Candidate aptamer evaluation for a given target protein.
+class BaseAptamerEval(BaseObject, ABC):
+    """Abstract base class for candidate aptamer evaluation against target proteins.
+
+    This class defines the common interface and shared functionality for aptamer
+    evaluation using different models or pipelines.
+
+    Parameters
+    ----------
+    target : str
+        Target sequence string.
+
+    Attributes
+    ----------
+    target : str
+        Target sequence string.
+    """
+
+    def __init__(self, target: str) -> None:
+        self.target = target
+        super().__init__()
+
+    def _inputnames(self) -> list[str]:
+        """Return the inputs of the experiment."""
+        return ["aptamer_candidate"]
+
+    def reconstruct(self, sequence: str = "") -> str:
+        """Reconstruct the actual aptamer sequence from the encoded representation.
+
+        The encoding uses pairs like 'A_' (add A to left) and '_A' (add A to right).
+        This method converts these pairs back to the actual sequence.
+
+        Parameters
+        ----------
+        sequence : str
+            Encoded sequence with direction markers (underscores).
+
+        Returns
+        -------
+        str
+            The reconstructed RNA sequence.
+        """
```

Comment:
Maybe add an example of what this method does with an incredibly small explanation just for clarity?